pallets/phala/src/migrations/mod.rs

```diff
@@ -120,3 +140,263 @@ pub mod v6 {
+				workers: VecDeque::new(),
+				cd_workers: VecDeque::new(),
+				lock_account,
+				owner_reward_account,
+			};
+			pool_info.workers.into_iter().for_each(|pubkey| {
+				new_pool_info.workers.push_back(pubkey);
+				let session: T::AccountId = stakepoolv2::pool_sub_account(pid, &pubkey);
+				let worker_info = computation::pallet::Sessions::<T>::get(&session)
+					.expect("session data should exist; qed.");
+				if worker_info.state == computation::pallet::WorkerState::WorkerCoolingDown {
+					new_pool_info.cd_workers.push_back(pubkey);
+				}
+				let computing_stake = computation::Stakes::<T>::get(&session).unwrap_or_default();
+				pawnshop::Pallet::<T>::mint_into(
+					&new_pool_info.owner_reward_account,
+					computing_stake,
+				)
+				.expect("mint into should be success");
+			});
+			pool_info
+				.withdraw_queue
+				.into_iter()
+				.for_each(|withdraw_info| {
+					let nft_id = basepool::Pallet::<T>::mint_nft(
+						collection_id,
+						basepool::pallet::pallet_id(),
+						withdraw_info.shares,
+					)
+					.expect("mint nft should always success");
+					new_pool_info
+						.basepool
+						.withdraw_queue
+						.push_back(basepool::WithdrawInfo {
+							user: withdraw_info.user,
+							start_time: withdraw_info.start_time,
+							nft_id,
+						});
+				});
+			let _ = computation::Pallet::<T>::withdraw_subsidy_pool(
+				&<T as pawnshop::Config>::PawnShopAccountId::get(),
+				pool_info.owner_reward,
+			);
+			// If the balance is too low to mint,we can just drop it.
+			pawnshop::Pallet::<T>::mint_into(
+				&new_pool_info.owner_reward_account,
+				pool_info.owner_reward,
+			);
+			pawnshop::Pallet::<T>::mint_into(
+				&new_pool_info.basepool.pool_account_id,
+				pool_info.free_stake,
+			);
+			basepool::pallet::Pools::<T>::insert(
+				pid,
+				poolproxy::PoolProxy::StakePool(new_pool_info),
+			);
+		});
+		basepool::PoolCount::<T>::put(stakepool::PoolCount::<T>::get());
+	}
+
+	fn migrate_pool_stakers<T: PhalaPallets>()
+	where
+		T: PhalaPallets + stakepool::pallet::Config,
+		BalanceOf<T>: balance_convert::FixedPointConvert + sp_std::fmt::Display,
+		T: pallet_uniques::Config<CollectionId = CollectionId, ItemId = NftId>,
+		T: pallet_assets::Config<AssetId = u32>,
+		T: pallet_assets::Config<Balance = BalanceOf<T>>,
+	{
+		stakepool::pallet::PoolStakers::<T>::drain().for_each(|((pid, user_id), staker_info)| {
+			let mut account_status = match pawnshop::StakerAccounts::<T>::get(&user_id) {
+				Some(account_status) => account_status,
+				None => pawnshop::FinanceAccount::<BalanceOf<T>> {
+					invest_pools: vec![],
+					locked: Zero::zero(),
+				},
+			};
+			let deprecated_pool = stakepool::pallet::StakePools::<T>::get(pid)
+				.expect("get deprecated pool should success; qed.");
+			let pending_reward = mul(staker_info.shares, &deprecated_pool.reward_acc.into())
+				- staker_info.reward_debt;
+			let user_reward = pending_reward + staker_info.available_rewards;
```

Comment:
May need to sample a few users and double check if the value is correct.

Reply:
We will add these test in post_migrate.